src/statement.d

```diff
@@ -3226,12 +3227,15 @@ public:
              * result of condition.
              */
             match = new VarDeclaration(loc, prm.type, prm.ident, new ExpInitializer(loc, condition));
-            match.parent = sc.func;
+            match.parent = scd.func;
             match.storage_class |= prm.storageClass;
+            match.semantic(scd);
```

Comment:
Is it obvious that `semantic` needs to be run on `match` separately from `condition.semantic`? I suspect that `VarExp` might store an incomplete type, but it would be nice to see a comment about this unexpected behaviour.